hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncTool.java

```diff
@@ -290,6 +290,9 @@ private boolean syncSchema(String tableName, boolean tableExists, boolean useRea
         // Sync the table properties if the schema has changed
         if (config.getString(HIVE_TABLE_PROPERTIES) != null || config.getBoolean(HIVE_SYNC_AS_DATA_SOURCE_TABLE)) {
           syncClient.updateTableProperties(tableName, tableProperties);
+          HoodieFileFormat baseFileFormat = HoodieFileFormat.valueOf(config.getStringOrDefault(META_SYNC_BASE_FILE_FORMAT).toUpperCase());
+          String serDeFormatClassName = HoodieInputFormatUtils.getSerDeClassName(baseFileFormat);
+          syncClient.updateTableSerDeInfo(tableName, serDeFormatClassName, serdeProperties);
```

Comment:
we don't need to pass serde class from the API. it's controlled by the base file format, which is taken from the sync config.